### DISCUSSION TITLE
Link verification fails between Mastodon instance and website on the same VPS

### DISCUSSION THREAD

**Discussion by janedenone** [mastodon/mastodon#19579] [Q&A] [answered]
Hi,

link verification between a Mastodon instance and a website running on the same VPS does not work "to prevent localhost rebinding attacks", according to [this thread](https://discourse.joinmastodon.org/t/profile-link-verification-fails/1752). Is this (still) true?

I would love to display the verified connection between social.mydomain.com and mydomain.com within my Mastodon profile, even though the connection is pretty obvious by itself.

Thanks,
Jan

**Comment by janedenone** [comment]
Update: I moved my website to a different VPS, but link verification still does not work between social.eden.one (Mastodon) and eden.one (website), although I placed 

```
<link rel="me" href="https://social.eden.one/@jan"/>
```

in the HTML head of the landing page (and added `https://eden.one` to my Mastodon profile links). I then tried delivering the website through a different domain (`eden.koeln`) – and the green verification checkmark appeared! So it seems impossible to verify links between the same base domain – why is that?

Thanks,
Jan

**Reply by janedenone** [reply]
Round 3: I identified the relevant code in `app/services/verify_link_service.rb`, `app/workers/verify_account_links_worker.rb` and `app/models/account/field.rb`. Nothing in these files seems to exclude the instance domain name from being verified. I am grateful for any hints.

Thanks,
Jan

**Comment by afontenot** [comment, marked as answer]
Total shot in the dark here: with certain resolver configurations, your domain name could get resolved locally to 127.0.0.1. So your Mastodon instance queries `eden.one`, gets the local address back. One possibility is that Mastodon rejects local IPs when DNS returns them. Probably not that unreasonable? Another possibility is that your webserver is not set up to listen on localhost.

Quick way to test this is to run `nslookup eden.one` on your server. If you get the localhost address back, there's a good chance this is your problem.

If so you could *probably* fix the problem by making sure that your local domain doesn't get resolved locally. E.g. change your hostname to be something other than `eden.one`, check the `/etc/hosts` file to see if it has any relevant entries, etc.

If it's not this, I have no clue, sorry!

**Reply by janedenone** [reply]
That did the trick! Thank you very much (and sorry for creating so much of a hassle for a negligible issue).

**Reply by afontenot** [reply]
No worries, that's what discussions are for!

**Reply by janedenone** [reply]
Embarassingly enough, I opened an issue with the Mastodon project in the meantime, being overly certain I had discovered a bug. :)

**Reply by Fywqdk** [reply]
Hi, I am hosting a small instance through masto.host on my own domain and have the same problem, but little knowledge of the tech behind and no access to /etc/hosts. Any hints or ideas for a way to circumvent this? (masto.epoch.dk / epoch.dk) Also hosted on one.com (which I assume eden.one is since it's their toplevel domain to my knowledge)

**Reply by janedenone** [reply]
Hi, no, eden.one is my own domain hosted somewhere else. I initially suspected that my problem was related to Mastodon and the website running on the same machine, and this might still be one possible cause. Could you redirect the website to a different server temporarily?

**Reply by Fywqdk** [reply]
Thanks for the quick response!
Ah ok. I will see if I can do that. Thanks!
